### PR TITLE
Javascript2 week1 Assignment

### PR DESCRIPTION
**Pull request by patel-prakashkumar**
I have finished JavaScript2 week1 Assignment. I haven't finished optional task - "Event types - Optional and a little tricky".

### CODE REVIEW

**Review by shravanipagilla — COMMENTED** [review]
Overall only one comment for your homeworkm.  💫 Great Work 💫
I don't have any more comments.
You did very hard and nice job Prakash.:)

File: javascript/javascript2/week1/main.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+function findShortestWord(danishWords) {
+    let shortest = danishWords.reduce((shortestWord, currentWord) => {
+        return currentWord.length < shortestWord.length ? currentWord : shortestWord;
+    }, danishWords[0]);
+    return shortest;
+}
+const shortestOutput = findShortestWord(danishWords);
+const displayOutput = document.createElement('h4')
+displayOutput.innerHTML = `Shortest word from List is : ${shortestOutput}`;
+shortestWord.appendChild(displayOutput);
```
Comment (MLMeyer) [inline javascript/javascript2/week1/main.js:16]: Looks good - great work 🔥

File: javascript/javascript2/week1/main.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        å: charÅ
+    }
+    return displyChar;
+}
+console.log(findDanishString(danishString))
+console.log(findDanishString(danishString2))
+const danishChar = JSON.stringify(findDanishString(danishString)) + JSON.stringify(findDanishString(danishString2));
+const displayCharOutput = document.createElement('p')
+displayCharOutput.innerHTML = `Results : ${danishChar}`
+danishLetter.appendChild(displayCharOutput);
```
Comment (MLMeyer) [inline javascript/javascript2/week1/main.js:42]: Good work! 

A detail to be aware of, is that the Danish letters can be repeated in the strings, in your variable danishString2, we in fact have 4 Danish letters with å: 1, æ: 1, and ø: 2. How can you update your implementation such that it includes the repeated Danish letters, both for the total count and for the individual letter count? 😊

File: javascript/javascript2/week1/main.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        const rating = document.createElement('li');
+        title.innerHTML = `Name: ${oneProduct.name}`
+        price.innerHTML = `Price: ${oneProduct.price}`
+        rating.innerHTML = `Rating: ${oneProduct.rating}`
+        productLi.appendChild(title).style.fontWeight = 'bold';
+        productLi.appendChild(price);
+        productLi.appendChild(rating);
+    }
+}
+renderProducts(products);
```
Comment (MLMeyer) [inline javascript/javascript2/week1/main.js:136]: Awesome work! 🔥

File: javascript/javascript2/week1/main.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+btnName.addEventListener('click', function () {
+    const randomName = Math.floor(Math.random() * spiritAnimal.length);
+    const text = inputName.value;
+    if (!text) {
+        displayName.innerHTML = 'Please Enter your Name'
+    }
+    else {
+        displayName.innerHTML = text + ' : ' + spiritAnimal[randomName];
+    }
+})
```
Comment (MLMeyer) [inline javascript/javascript2/week1/main.js]: Great implementation - I like your choice of spirit animal names! 🐜🦊🐐🦒

File: javascript/javascript2/week1/main.js (hunk 2 above)
Comment (MLMeyer) [inline javascript/javascript2/week1/main.js:46]: Good work! 

A detail to be aware of, is that the Danish letters can be repeated in the strings, in your variable danishString2, we in fact have 4 Danish letters with å: 1, æ: 1, and ø: 2. How can you update your implementation such that it includes the repeated Danish letters, both for the total count and for the individual letter count? 😊

**Review by MLMeyer — CHANGES_REQUESTED** [review]

File: javascript/javascript2/week1/main.js (hunk 1 above)
Comment (patel-prakashkumar) [reply javascript/javascript2/week1/main.js:16]: Thank you for feedback.

File: javascript/javascript2/week1/main.js (hunk 2 above)
Comment (patel-prakashkumar) [reply javascript/javascript2/week1/main.js:46]: Yes, you are right, Its showing wrong output.  I have fixed issue other way. I have some doubt that when I using "match(/å/i)", its showing wrong out and when I am using "match(/å/g)" showing correct output. I have used  keyword "i" for capital and small character identify. why its not working as suppose to work?

**Pull request merged**